pkg/install/start_vms.go

```diff
@@ -0,0 +1,55 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+)
+
+// startVMs checks cluster VMs power state and starts deallocated VMs (if any)
+func (i *Installer) startVMs(ctx context.Context) error {
+	resourceGroupName := stringutils.LastTokenByte(i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	// TODO: Check if customer can create VMs in the cluster resource group (deny assignment?)
+	vms, err := i.virtualmachines.List(ctx, resourceGroupName)
+	if err != nil {
+		return err
+	}
+
+	deallocatedVMs := make([]mgmtcompute.VirtualMachine, 0, len(vms))
+	for _, vm := range vms {
+		// TODO: consider getting instance view concurrently with errgroup
+		instanceView, err := i.virtualmachines.InstanceView(ctx, resourceGroupName, *vm.Name)
```

Comment:
fwiw pkg/frontend/admin_openshiftcluster_resources_list.go does vmClient.Get(ctx, clusterResourceGroup, *res.Name, mgmtcompute.InstanceView) to much the same effect